src/setupTests.js

```diff
@@ -1,4 +1,78 @@
+import React from 'react'
+import { act } from 'react-dom/test-utils'
 import Adapter from 'enzyme-adapter-react-16'
-import { configure } from 'enzyme'
+import { configure, mount } from 'enzyme'
+import 'jest-localstorage-mock'
+
+import App from './App.js'
+import { keyDown } from './shortcuts.js'
 
 configure({ adapter: new Adapter() })
+
+const createTestApp = async app => {
+  document.getSelection = () => ({
+    type: 'None',
+    removeAllRanges: () => {},
+  })
+
+  document.createRange = () => ({
+    setStart: () => {},
+    setEnd: () => {},
+    commonAncestorContainer: {
+      nodeName: 'BODY',
+      ownerDocument: document,
+    },
+    collapse: () => {},
+  })
+
+  window.getSelection = () => ({
+    focusOffset: 3,
+    removeAllRanges: () => {},
+    addRange: () => {},
+  })
+
+  window.location = {
+    pathname: '',
+  }
+
+  await act(async () => {
+    jest.useFakeTimers()
+    const originalError = console.error
```

Comment:
Hey, I'm not really comfortable with disabling `console.error` to hide the error. It seems like it's just a workaround. Let's try to get to the source of the problem. If it is impossible to remove, it deserves a clear comment explaining why.

Reply:
I think https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMLegacy.js#L81 is where we get that error from. As I don't see any way to go around it except creating new `console.error` which filters what it outputs by content (we can output only if first argument to `console.error` is not the one from the link).

Reply:
New `console.error` mock is more precise in what it silences. Good enough?

Reply:
Mounting to a child of `document.body` will fix the error without the workaround:

```
    // mount to child to avoid warning about rendering directly to document.body
    const root = document.body.appendChild(document.createElement('div'))

    const wrapper = await mount(
      <div
        id="keyboard"
        onKeyDown={keyDown}
        tabIndex="0"
      >
        {app}
      </div>,
      { attachTo: root },
    )
```